Pull Request for Full Review

PlagueHO opened the pull request:
PR for performing a full module review.

See: https://github.com/PowerShell/DscResources/issues/75

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/plagueho/wsmandsc/1)

<!-- Reviewable:end -->

kwirkykat commented:
Reviewed 2 of 21 files at r1.
Review status: 2 of 21 files reviewed at latest revision, 6 unresolved discussions.

---

*[appveyor.yml, line 11 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOvOk6DaJvVNB5OqK4d:-KOvOk6DaJvVNB5OqK4e:b-r4adx) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/appveyor.yml#L11)):*
> ```YAML
>         cd DscResource.Tests
>         Import-Module .\TestHelper.psm1 -force
>         Pop-Location
> ```

Don't need the change of location here
```powershell
Import-Module .\DscResource.Tests\TestHelper.psm1 -Force
```

---

*[LICENSE, line 21 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOvOzyLrD9gnidyXxef:-KOvOzyLrD9gnidyXxeg:b-tfxnfu) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/LICENSE#L21)):*
> ```
> LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
> OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
> THE SOFTWARE.
> ```

Missing a newline

---

*[README.md, line 6 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOvP2J8xl__GacItW_P:-KOvP2J8xl__GacItW_Q:bkm9c42) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/README.md#L6)):*
> ```Markdown
> 
> The **WSManDsc** module contains DSC resources for configuring WS-Management and PowerShell Remoting.
> ```

Please add snippet about the MS open source code of conduct:
```
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
```

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 62 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOvNr813tzQKh2C8ChS:-KOvNr813tzQKh2C8ChT:bqkaodw) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L62)):*
> ```PowerShell
>             URLPrefix = $Listener.URLPrefix
>             CertificateThumbprint = $Listener.CertificateThumbprint
>             }
> ```

This bracket needs to go back one indent

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 368 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOw5iKoHEKn13nr5Izj:-KOveXmssrOWFQr5rUX1:b-6gqkmw) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L368)):*
> ```PowerShell
> } # Test-TargetResource
> 
> function Get-Listener
> ```

Please add a help comment above this function.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 390 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOw5hDXNsYgubi4Blyz:-KOveK1C5SM6Sgw8aFtl:b-6gqkmw) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L390)):*
> ```PowerShell
> } # Get-Listener
> 
> function Get-DefaultPort
> ```

Please add a help comment above this function.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Reviewed 4 of 21 files at r1.
Review status: 4 of 21 files reviewed at latest revision, 13 unresolved discussions.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 19 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP06xT7oConSGDpQ91m:-KP06xT7oConSGDpQ91n:b-6sr1xz) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L19)):*
> ```PowerShell
> #endregion
> ```

Add in Help-comments

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 21 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP02NQ1PSQbxYSgKywf:-KP02NQ1PSQbxYSgKywg:bfml6s7) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L21)):*
> ```PowerShell
> 
> function Get-TargetResource
> {
> ```

Add:
```powershell
[CmdletBinding()]
```
above the param blocks

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 146 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0D2eGsakEMGhnxiow:-KP0D2eHj0gpBf_nOrMd:bt8ysig) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L146)):*
> ```PowerShell
>                     -f $Transport,$Port
>                 ) -join '' )
>             # Ths listener doesn't exist - do nothing
> ```

move this comment above Write-Verbose

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 210 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0DmlBNR9V8-xsuD2x:-KP0DmlC11ze3P_PSJOv:bz0xiop) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L210)):*
> ```PowerShell
>             }
>             else
>             {
> ```

Could you add comments at the top of these else statements as well? Easy to get lost in all of the conditionals...

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 255 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0E43CVw-estvHmbv2:-KP0E43CVw-estvHmbv3:b9ljdl5) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L255)):*
> ```PowerShell
> 
> function Test-TargetResource
> {
> ```

Again, add the cmdlet binding here as well

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 295 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0ERgdms6ZLOBoSavM:-KP0ERgdms6ZLOBoSavN:b-cswsim) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L295)):*
> ```PowerShell
> 
>     # Lookup the existing Listener
>     $Listener = Get-Listener -Transport $Transport
> ```

change the local variables to be lowercase - $listener and $port

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 409 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0FBdTtBMqis9w4yLB:-KP0FBdTtBMqis9w4yLC:begh4sv) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L409)):*
> ```PowerShell
>         if ($Transport -eq 'HTTP')
>         {
>             $Port = 5985
> ```

Can we declare these as constants above? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Review status: 2 of 21 files reviewed at latest revision, 13 unresolved discussions.

---

*[appveyor.yml, line 11 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOvOk6DaJvVNB5OqK4d:-KP4QOLKsmln4_6zfxfd:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/appveyor.yml#L11)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Don't need the change of location here
> ```powershell
> Import-Module .\DscResource.Tests\TestHelper.psm1 -Force
> ```
</details>
Done.

---

*[LICENSE, line 21 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOvOzyLrD9gnidyXxef:-KP4QTcpNXBARxvC0-be:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/LICENSE#L21)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Missing a newline
</details>
Done.

---

*[README.md, line 6 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOvP2J8xl__GacItW_P:-KP4QaxTiO_H6Nh3LvrD:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/README.md#L6)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please add snippet about the MS open source code of conduct:
> ```
> This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
> For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
> ```
</details>
Done.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 19 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP06xT7oConSGDpQ91m:-KP4mMfqC2xDnUYtUsbP:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L19)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Add in Help-comments
</details>
Done.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 21 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP02NQ1PSQbxYSgKywf:-KP4md4BSxgDh53Q4EW3:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L21)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Add:
> ```powershell
> [CmdletBinding()]
> ```
> above the param blocks
</details>
Done.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 62 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOvNr813tzQKh2C8ChS:-KP4mipophOds6aGTLPN:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L62)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> This bracket needs to go back one indent
</details>
Done.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 146 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0D2eGsakEMGhnxiow:-KP4mqkznXVDmBLiMem_:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L146)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> move this comment above Write-Verbose
</details>
Done.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 210 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0DmlBNR9V8-xsuD2x:-KP4oOQz8EbOZ6mP4QhF:b-x81bhs) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L210)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Could you add comments at the top of these else statements as well? Easy to get lost in all of the conditionals...
</details>
Done. I also fixed the >100 char lines.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 255 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0E43CVw-estvHmbv2:-KP4oYQ4D-AvxCuL168j:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L255)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Again, add the cmdlet binding here as well
</details>
Done.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 295 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0ERgdms6ZLOBoSavM:-KP4okqc5v6ydTt5G9aA:b-4h7c74) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L295)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> change the local variables to be lowercase - $listener and $port
</details>
I've done the $listener variable (and some other ones), but I didn't do $Port - because it is actually a Parameter. Should I change it or leave it as $Port because it does get passed in?

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 368 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOw5iKoHEKn13nr5Izj:-KP4pLjA8kz73cCTU1K_:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L368)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please add a help comment above this function.
</details>
Done.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 390 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KOw5hDXNsYgubi4Blyz:-KP4pMO33gsVuuapH_0T:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L390)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please add a help comment above this function.
</details>
Done.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 409 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0FBdTtBMqis9w4yLB:-KP4q6t0NtPWmqTDJ7NF:bz5py0o) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L409)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Can we declare these as constants above? 
</details>
Good idea! Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
I've made the following additional changes (based on your reviews of SystemLocaleDsc):
- Updated unit test/integration test headers and replaced $Global: with $Script
- Added description of examples to example files
- Added Get-Help to all functions in MSFT_WSManServiceConfig.
- Fixed up 100 char limit violations (the ruler in VS Code is amazing for this if you're interested).
- Added Fixed local variable name capitalization ($Parameter -> $parameter) in MSFT_WSManServiceConfig
- Added spaces around equals signs in each item in Localization files.

---

Review status: 2 of 21 files reviewed at latest revision, 13 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-:-KP52HVmvM3ynVyNwGC8:b-pq97h0)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Reviewed 1 of 21 files at r1.
Review status: 3 of 21 files reviewed at latest revision, 18 unresolved discussions.

---

*[DSCResources/MSFT_WSManListener/en-us/MSFT_WSManListener.psd1, line 2 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0GaCJwhvypFC7-ok4:-KP0GaCKK0wiHMhrfP-m:b-oqz9oq) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/en-us/MSFT_WSManListener.psd1#L2)):*
> ```PowerShell
> ConvertFrom-StringData @'
> GettingListenerMessage=Getting Listener.
> ```

indent the lines within the string

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.parameterlist.psd1, line 5 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0GnDusenYpjHgkH1M:-KP0GnDusenYpjHgkH1N:b-tlgc3v) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.parameterlist.psd1#L5)):*
> ```PowerShell
>            Path    = 'RootSDDL';
>            Type    = 'String';
>            Default = 'O:NSG:BAD:P(A;;GA;;;BA)(A;;GR;;;IU)S:P(AU;FA;GA;;;WD)(AU;SA;GXGW;;;WD)';
> ```

Is this supposed to be GXGW with no ';' in between? I don't have enough background on this to know...

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1, line 30 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0Hm_RqJANtUHFcQiZ:-KP0Hm_RqJANtUHFcQi_:b-puq1jd) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1#L30)):*
> ```PowerShell
>     -Path $PSScriptRoot `
>     -ChildPath 'MSFT_WSManServiceConfig.parameterlist.psd1'
> $ParameterList = Invoke-Expression "DATA { $(Get-Content -Path $ParameterListPath -Raw) }"
> ```

Would it be possible to refactor this to not use Invoke-Expression? It's okay in some places like integration tests - but if there's a relatively simple work around, that's preferable. 

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1, line 70 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPE7c6KWj_dzgirHLfs:-KPE7c6KWj_dzgirHLft:b-t44txi) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1#L70)):*
> ```PowerShell
>     (
>         [parameter(Mandatory = $true)]
>         [ValidateSet('Yes')]
> ```

Does this mean it can only ever be set to 'Yes'?

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1, line 131 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPE7tcWB5GtAudDPJis:-KPE7tcWB5GtAudDPJit:b-ccwzhu) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1#L131)):*
> ```PowerShell
> 
>         $ParameterCurrent = (Get-Item -Path $ParameterPath).Value
>         $ParameterNew = (Invoke-Expression -Command "`$$($Parameter.Name)")
> ```

Again, Invoke-Expression should be replaced here

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Reviewed 2 of 21 files at r1.
Review status: 4 of 21 files reviewed at latest revision, 32 unresolved discussions.

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.schema.mof, line 4 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPE8bxGQVCE7UVRNo18:-KPE8bxGQVCE7UVRNo19:b-jkanw4) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.schema.mof#L4)):*
> ```
> class MSFT_WSManServiceConfig : OMI_BaseResource
> {
>     [Key, Description("Specifies the resource is a single instance, the value must be 'Yes'"), ValueMap{"Yes"}, Values{"Yes"}] String IsSingleInstance;
> ```

if this has to be yes, I don't think it should be a parameter - have a different check for this. Is there anything else that can be used as the key?

---

*[Examples/Sample_WSManListener_HTTP.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPE9yvX8XUUKSz_4KF9:-KPE9yvX8XUUKSz_4KFA:b1jn7ww) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Examples/Sample_WSManListener_HTTP.ps1#L1)):*
> ```PowerShell
> configuration Sample_WSManListener_HTTP
> ```

Can you add a short intro comment, briefly explaining when to use this example and for what?

---

*[Examples/Sample_WSManListener_HTTPS.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEA7RV4Azu3APxbKTZ:-KPEA7RV4Azu3APxbKT_:b-b0mvri) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Examples/Sample_WSManListener_HTTPS.ps1#L1)):*
> ```PowerShell
> configuration Sample_WSManListener_HTTPS
> ```

Same as above - add a comment

---

*[Examples/Sample_WSManServiceConfig.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEABi-7Ck6RqLv1ERT:-KPEABi-7Ck6RqLv1ERU:bo3pjvp) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Examples/Sample_WSManServiceConfig.ps1#L1)):*
> ```PowerShell
> configuration Sample_WSManServiceConfig
> ```

Add comment 

---

*[Tests/Integration/MSFT_WSManListener.Integration.Tests.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEAQPLzRD9JUVbTW5d:-KPEAQPLzRD9JUVbTW5e:bwddeos) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Integration/MSFT_WSManListener.Integration.Tests.ps1#L1)):*
> ```PowerShell
> $Global:DSCModuleName   = 'WSManDsc'
> ```

change to $script: variables

---

*[Tests/Integration/MSFT_WSManServiceConfig.Integration.Tests.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEB7RnItq4HuevrH6a:-KPEB7RnItq4HuevrH6b:b-utdmfk) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Integration/MSFT_WSManServiceConfig.Integration.Tests.ps1#L1)):*
> ```PowerShell
> $Global:DSCModuleName   = 'WSManDsc'
> ```

$script:

---

*[Tests/Unit/MSFT_WSManListener.Tests.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEBPRbcYi8_OvLfLFM:-KPEBPRbcYi8_OvLfLFN:b-jbopme) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManListener.Tests.ps1#L1)):*
> ```PowerShell
> $Global:DSCModuleName   = 'WSManDsc'
> ```

$script

---

*[Tests/Unit/MSFT_WSManListener.Tests.ps1, line 23 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPECldXpHUwxYMU1HHu:-KPECldXpHUwxYMU1HHv:b-9jarcz) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManListener.Tests.ps1#L23)):*
> ```PowerShell
> try
> {
>     # Make sure WS-Man is senabled
> ```

'is enabled' - typo

---

*[Tests/Unit/MSFT_WSManListener.Tests.ps1, line 130 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEBvNseYbiljuPm79X:-KPEBvNt2mwe2oIx-iqF:b-7egsb9) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManListener.Tests.ps1#L130)):*
> ```PowerShell
> 
>                 Mock Get-WSManInstance -MockWith { }
>                 Mock Remove-WSManInstance -MockWith { }
> ```

Any of these mocks that are being mocked the same in each Context block can be moved up into the Describe block and only declared onece

---

*[Tests/Unit/MSFT_WSManListener.Tests.ps1, line 225 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPECKbSagc1cGudmTQZ:-KPECKbSagc1cGudmTQ_:b-61oufi) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManListener.Tests.ps1#L225)):*
> ```PowerShell
>             }
> 
>             Context 'Both Listeners exists and HTTPS should' {
> ```

can you rephrase this context? It's a little confusing

---

*[Tests/Unit/MSFT_WSManListener.Tests.ps1, line 332 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEC_ZgkqmOdzJI_MXZ:-KPEC_ZgkqmOdzJI_MX_:b-4fphw3) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManListener.Tests.ps1#L332)):*
> ```PowerShell
>             }
> 
>             Context 'Both Listeners exists and HTTPS should' {
> ```

Again, can you rephrase this?

---

*[Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPECdO0iQA-Rlo3NP-D:-KPECdO0iQA-Rlo3NP-E:b-utdmfk) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1#L1)):*
> ```PowerShell
> $Global:DSCModuleName   = 'WSManDsc'
> ```

$script:

---

*[Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1, line 29 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEChD7TJxKHFCi-_zF:-KPEChD7TJxKHFCi-_zG:b-9jarcz) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1#L29)):*
> ```PowerShell
> try
> {
>     # Make sure WS-Man is senabled
> ```

'is enabled' - typo

---

*[Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1, line 198 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEDCXdOvqUK0IFJSeh:-KPEDCXdOvqUK0IFJSei:bx4l3hi) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1#L198)):*
> ```PowerShell
>             Context 'Create a TestError Exception' {
> 
>                 It 'should throw an TestError exception' {
> ```

'a' instead of 'an'

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Reviewed 4 of 15 files at r2.
Review status: 8 of 21 files reviewed at latest revision, 26 unresolved discussions.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1, line 295 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0ERgdms6ZLOBoSavM:-KPF-z-qvlbVmMfI1qol:b-akc5ot) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/MSFT_WSManListener.psm1#L295)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary>
> I've done the $listener variable (and some other ones), but I didn't do $Port - because it is actually a Parameter. Should I change it or leave it as $Port because it does get passed in?
</details>
LGTM - Yeah, sorry about that. $Port should stay as is

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Review status: 8 of 21 files reviewed at latest revision, 25 unresolved discussions.

---

*[DSCResources/MSFT_WSManListener/en-us/MSFT_WSManListener.psd1, line 2 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0GaCJwhvypFC7-ok4:-KPHEXGAQ-iIO5O4PRpT:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManListener/en-us/MSFT_WSManListener.psd1#L2)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> indent the lines within the string
</details>
Done.

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.parameterlist.psd1, line 5 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0GnDusenYpjHgkH1M:-KPHEdnBRIkey-Aci6ub:b-gbdmiw) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.parameterlist.psd1#L5)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Is this supposed to be GXGW with no ';' in between? I don't have enough background on this to know...
</details>
I think it's correct. Multiple permissions can be assigned by appending them together: https://itconnect.uw.edu/wares/msinf/other-help/understanding-sddl-syntax/

SDDL does my head in. 

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1, line 30 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0Hm_RqJANtUHFcQiZ:-KPHOsSfFoCAuxVLx0e1:bcboc2o) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1#L30)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Would it be possible to refactor this to not use Invoke-Expression? It's okay in some places like integration tests - but if there's a relatively simple work around, that's preferable. 
</details>
Leave this one with me. I'll see if I can find an alternate approach to this.

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1, line 70 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPE7c6KWj_dzgirHLfs:-KPHJM_Tb9uIi0tfE7K8:b-z1mip5) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1#L70)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Does this mean it can only ever be set to 'Yes'?
</details>
Yes. It does. This is the standard Single Instance design pattern Travis came up with a while back (https://blogs.msdn.microsoft.com/powershell/2015/06/22/want-to-write-a-dsc-resource-where-only-a-single-instance-can-be-configured/)

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1, line 131 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPE7tcWB5GtAudDPJis:-KPHOzzSw2ld6dkYmuRU:bcboc2o) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1#L131)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Again, Invoke-Expression should be replaced here
</details>
Leave this one with me. I'll see if I can find an alternate approach to this.

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.schema.mof, line 4 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPE8bxGQVCE7UVRNo18:-KPHKZTiT2DHZf3fUTBU:b-6jplw8) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.schema.mof#L4)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> if this has to be yes, I don't think it should be a parameter - have a different check for this. Is there anything else that can be used as the key?
</details>
See my comments above about the standard pattern :smile:

---

*[Examples/Sample_WSManListener_HTTP.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPE9yvX8XUUKSz_4KF9:-KPHLrtxfvSvV9hhFD6C:b-n95qzr) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Examples/Sample_WSManListener_HTTP.ps1#L1)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Can you add a short intro comment, briefly explaining when to use this example and for what?
</details>
Doh - I think I must have made these changes before you published your comments as I'd already added comments to the examples based on the review you performed on xSystemLocale ! Anyway, done :smile:

---

*[Examples/Sample_WSManListener_HTTPS.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEA7RV4Azu3APxbKTZ:-KPHLtCWCO_7SeKJecx1:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Examples/Sample_WSManListener_HTTPS.ps1#L1)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Same as above - add a comment
</details>
Done.

---

*[Examples/Sample_WSManServiceConfig.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEABi-7Ck6RqLv1ERT:-KPHMDKfph5FsMoI5vlR:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Examples/Sample_WSManServiceConfig.ps1#L1)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Add comment 
</details>
Done.

---

*[Tests/Integration/MSFT_WSManListener.Integration.Tests.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEAQPLzRD9JUVbTW5d:-KPHMfzlGG1wIkfpfIa4:b-quv9vk) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Integration/MSFT_WSManListener.Integration.Tests.ps1#L1)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> change to $script: variables
</details>
Had already done this one too :smile: I've been keeping a running list of mistakes I make and trying to fix all my resources before you get to them. So, done.

---

*[Tests/Integration/MSFT_WSManServiceConfig.Integration.Tests.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEB7RnItq4HuevrH6a:-KPHMv9vPisCkBIyn9wg:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Integration/MSFT_WSManServiceConfig.Integration.Tests.ps1#L1)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> $script:
</details>
Done.

---

*[Tests/Unit/MSFT_WSManListener.Tests.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEBPRbcYi8_OvLfLFM:-KPHMvg1gYxXfBOW90bG:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManListener.Tests.ps1#L1)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> $script
</details>
Done.

---

*[Tests/Unit/MSFT_WSManListener.Tests.ps1, line 23 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPECldXpHUwxYMU1HHu:-KPHN-jWh9ujm3-Wj-3l:b-o7vuhc) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManListener.Tests.ps1#L23)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'is enabled' - typo
</details>
Good catch! Done.

---

*[Tests/Unit/MSFT_WSManListener.Tests.ps1, line 130 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEBvNseYbiljuPm79X:-KPHO-mBuUgldZMwoCva:bcvjgzp) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManListener.Tests.ps1#L130)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Any of these mocks that are being mocked the same in each Context block can be moved up into the Describe block and only declared onece
</details>
Good idea. Done.

---

*[Tests/Unit/MSFT_WSManListener.Tests.ps1, line 225 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPECKbSagc1cGudmTQZ:-KPHOZ3YDa6i1o-aUvh2:b-wm6qbs) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManListener.Tests.ps1#L225)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> can you rephrase this context? It's a little confusing
</details>
Definitely. Done.

---

*[Tests/Unit/MSFT_WSManListener.Tests.ps1, line 332 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEC_ZgkqmOdzJI_MXZ:-KPHObCGoKB9lDy8_DZt:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManListener.Tests.ps1#L332)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Again, can you rephrase this?
</details>
Done.

---

*[Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPECdO0iQA-Rlo3NP-D:-KPHOcY7KwKn5D5wcG26:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1#L1)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> $script:
</details>
Done.

---

*[Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1, line 29 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEChD7TJxKHFCi-_zF:-KPHOiNPS4CPuD7Sj9uI:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1#L29)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'is enabled' - typo
</details>
Done.

---

*[Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1, line 198 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPEDCXdOvqUK0IFJSeh:-KPHOqbUVq-FBC67Su8n:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/Tests/Unit/MSFT_WSManServiceConfig.Tests.ps1#L198)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'a' instead of 'an'
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Review status: 8 of 21 files reviewed at latest revision, 25 unresolved discussions.

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1, line 30 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KP0Hm_RqJANtUHFcQiZ:-KPHVBz4qE2cXB-UdXuZ:b-7p9tpa) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1#L30)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary>
> Leave this one with me. I'll see if I can find an alternate approach to this.
</details>
Ok. I've changed the pattern to get rid of the Invoke-Expression. So this should now be good to go.

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1, line 131 \[r1\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KPE7tcWB5GtAudDPJis:-KPHVK5h9OnTwMonyrAZ:b-7p9tpa) ([raw file](https://github.com/plagueho/wsmandsc/blob/de39aec3eb7c6a98bd05634d325f2321c22e0bf8/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.psm1#L131)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary>
> Leave this one with me. I'll see if I can find an alternate approach to this.
</details>
Ok. I've changed the pattern to get rid of the Invoke-Expression. So this should now be good to go.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 21 files at r1, 11 of 11 files at r3.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-:-KPJVYATTOdYjwbUCR63:bnfp4nl)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
@kwirkykat - I think this one should be all ready to go now (hopefully :smile: )

---

Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-:-KPX2tvnTqY4dl2lHB02:bp5ipvs)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.schema.mof, line 4 \[r3\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KQ78-9bVcCkL6LlVSXu:-KQ8cOamyBWF_H4KPpfU:b-43szpk) ([raw file](https://github.com/plagueho/wsmandsc/blob/61dd8d6e018e7bbacebd23210d20190e65569df5/DSCResources/MSFT_WSManListener/MSFT_WSManListener.schema.mof#L4)):*
<details><summary><i>Previously, TravisEz13 (Travis Plunk) wrote…</i></summary>
> Compatibility​ listener is not here, but this would not be a breaking change.
</details>
I've been doing digging around in this:

The Compatibility Listeners behave and are configured differently to a standard WSMan Listener. We don't actually create them via the New-WSManInstance (and can't - so they can't be reconfigured by removing and recreating them). They have to be created by setting the WSMan:\Localhost\Service\EnableCompatibilityHTTPListener setting to true. You also can't change the ports of the compatibility listeners either. Basically compatibility listeners can only be disabled or enabled.

If enabling compatibility listeners was performed by WSManListener then I'd need to remove the ability from WSManServiceConfig as this could lead to a conflict in configs. This would be a breaking change (that is fine though).

Also, I would recommend that if we were to move the ability to enable/disable the compatibility listeners to this resource then we should add a new optional bool parameter "Compatibility" (that defaults to false) - rather than adjusting the Transport ("Compatibility" is not really a transport and would be counter intuitive).

But IMHO we really shouldn't be encouraging people to use compatibility listeners. Shouldn't we be discouraging their use? I wrote the original ability into WSManServiceConfig because at the enterprise I work, we had 100s of servers floating around where someone had randomly enabled them and our team wanted to ensure that this was never the case (it causes all sorts of problems and headaches and governance issues).

I'll take your guidance on this.

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.schema.mof, line 8 \[r3\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KQ792cyGqgTM-B8SEuW:-KQ8-TNjUlyRAec_FrU5:b-896fix) ([raw file](https://github.com/plagueho/wsmandsc/blob/61dd8d6e018e7bbacebd23210d20190e65569df5/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.schema.mof#L8)):*
<details><summary><i>Previously, TravisEz13 (Travis Plunk) wrote…</i></summary>
> Capitalize `ms` or spell it out
</details>
Done.

---

*[DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.schema.mof, line 16 \[r3\]](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KQ79MAM7YzIwry4_nb4:-KQ7zvklYvD8jHu5l_Zr:b6rex45) ([raw file](https://github.com/plagueho/wsmandsc/blob/61dd8d6e018e7bbacebd23210d20190e65569df5/DSCResources/MSFT_WSManServiceConfig/MSFT_WSManServiceConfig.schema.mof#L16)):*
<details><summary><i>Previously, TravisEz13 (Travis Plunk) wrote…</i></summary>
> I need to follow up here.  I don't understand want CBT is.
</details>
TBH, neither do I. It was the only one I wasn't 100% familiar with. I exposed it to try and be consistent across the resource.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
@kwirkykat - are you guys waiting on something more from me on this one? I want to try and close some of my back log items so I can work on some newer stuff :smile:

---

Review status: 16 of 21 files reviewed at latest revision, 9 unresolved discussions.

---

*[DSCResources/MSFT_WSManListener/MSFT_WSManListener.schema.mof, line 4 at r3](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-KQ78-9bVcCkL6LlVSXu:-KRXqgGTTP_7qABiDB2_:bh04u9a) ([raw file](https://github.com/plagueho/wsmandsc/blob/61dd8d6e018e7bbacebd23210d20190e65569df5/DSCResources/MSFT_WSManListener/MSFT_WSManListener.schema.mof#L4)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary>
> I've been doing digging around in this:
> 
> The Compatibility Listeners behave and are configured differently to a standard WSMan Listener. We don't actually create them via the New-WSManInstance (and can't - so they can't be reconfigured by removing and recreating them). They have to be created by setting the WSMan:\Localhost\Service\EnableCompatibilityHTTPListener setting to true. You also can't change the ports of the compatibility listeners either. Basically compatibility listeners can only be disabled or enabled.
> 
> If enabling compatibility listeners was performed by WSManListener then I'd need to remove the ability from WSManServiceConfig as this could lead to a conflict in configs. This would be a breaking change (that is fine though).
> 
> Also, I would recommend that if we were to move the ability to enable/disable the compatibility listeners to this resource then we should add a new optional bool parameter "Compatibility" (that defaults to false) - rather than adjusting the Transport ("Compatibility" is not really a transport and would be counter intuitive).
> 
> But IMHO we really shouldn't be encouraging people to use compatibility listeners. Shouldn't we be discouraging their use? I wrote the original ability into WSManServiceConfig because at the enterprise I work, we had 100s of servers floating around where someone had randomly enabled them and our team wanted to ensure that this was never the case (it causes all sorts of problems and headaches and governance issues).
> 
> I'll take your guidance on this.
> 
</details>
Hi @travisez13  - have you had a chance to think about this one? I'd love to get this resource merged over and off my ever growing backlog :smile:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/wsmandsc/1#-:-KRXquRBpXv-VYCytOIK:b-58tsxz)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Sorry for the delay. Travis is currently out of office, but hopefully we can get this review finished ASAP when he gets back.

kwirkykat commented:
@PlagueHO Nope, not waiting on you. Sorry. We're just swamped.

PlagueHO commented:
@kwirkykat, @mbreakey3 - Just a heads up: I've had a request for a new feature to be added to this resource. I have added it but also did a bit of refactoring to improve testability - so I also added a bunch more unit tests. So this new feature (adding DN parameter to WSManListener) is included in this PR now.
